### PR TITLE
Clarify documentation and suggest using RelWithDebInfo.

### PR DESCRIPTION
**Pull request by ScottTodd**
* Clarify unchecked "supported" platforms/configs (this is still fuzzy - open to suggestions)
* Rename "Linux" to "Linux and MacOS" in a few places
* Suggest `-DCMAKE_BUILD_TYPE=RelWithDebInfo` in CMake configure commands

### CODE REVIEW

File: docs/website/docs/building-from-source/getting-started.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ git submodule update --init
 Configure then build all targets using CMake:
 
 ``` shell
-cmake -B ../iree-build/ .
+cmake -B ../iree-build/ -DCMAKE_BUILD_TYPE=RelWithDebInfo .
```
Comment (stellaraccident) [inline docs/website/docs/building-from-source/getting-started.md:48]: I feel like this needs a tip below which says that this disables various debugging features (`-debug`, asserts) and if those are valuable, building with `Debug` is recommended, at the expense of the resulting compiler and runtime being significantly slower.

(really wish we could enable those in release mode, so maybe can update this guidance at some point)

File: docs/website/docs/index.md [hunk 2]
```diff
@@ -23,20 +23,20 @@ IREE supports importing from a variety of ML frameworks:
 - [x] TensorFlow
 - [x] TensorFlow Lite
 - [x] JAX
-- [ ] PyTorch
-<!-- - [ ] ONNX -->
+- [ ] PyTorch (planned)
+<!-- - [ ] ONNX (planned) -->
```
Comment (stellaraccident) [inline docs/website/docs/index.md:27]: Is "hoped" a category (as a step below "planned") :)

**Review by stellaraccident — APPROVED** [review]

File: docs/website/docs/index.md (hunk 2 above)
Comment (ScottTodd) [reply docs/website/docs/index.md]: Sure :P uncommented and changed to "(hoped for)"

File: docs/website/docs/building-from-source/getting-started.md (hunk 1 above)
Comment (ScottTodd) [reply docs/website/docs/building-from-source/getting-started.md:48]: SG. Added a tip:

```
???+ Tip
    We recommend using the `RelWithDebInfo` build type by default for a good
    balance of debugging information and performance. The `Debug`, `Release`,
    and `MinSizeRel` build types are useful in more specific scenarios. See the
    [official CMake documentation](https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html)
    for more details.
```

Comment (stellaraccident) [reply docs/website/docs/building-from-source/getting-started.md:48]: Possibly also "note that several useful LLVM debugging features are only available in Debug builds." Nothing like expecting to find certain features and not knowing how to get them (ie. `-debug`)

**Pull request merged**